// react-multilevel-dropdown/DropdownUsage.stories.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.OpenOnHover = exports.CloseOnItemClick = exports.OpenProgrammatically = exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _index = _interopRequireDefault(require("./index"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var _default = {
  title: 'Dropdown Usages'
};
exports.default = _default;

var OpenProgrammatically = function OpenProgrammatically() {
  var ref = (0, _react.useRef)(null);

  function onClick(e) {
    if (ref.current) {
      ref.current.toggle(e);
    }
  }

  return _react.default.createElement("div", {
    style: {
      marginLeft: 400
    }
  }, _react.default.createElement("button", {
    onClick: onClick,
    style: {
      marginRight: 100
    }
  }, "open dropdown"), _react.default.createElement(_index.default, {
    title: "Dropdown example",
    ref: ref
  }, _react.default.createElement(_index.default.Item, null, "Item 1"), _react.default.createElement(_index.default.Item, null, "Item 2"), _react.default.createElement(_index.default.Item, null, "Item 3"), _react.default.createElement(_index.default.Item, null, "Item 4 with sub", _react.default.createElement(_index.default.Submenu, null, _react.default.createElement(_index.default.Item, null, "Sub item 1 with sub", _react.default.createElement(_index.default.Submenu, null, _react.default.createElement(_index.default.Item, null, "Sub Sub 1"), _react.default.createElement(_index.default.Item, null, "Sub Sub 2"))), _react.default.createElement(_index.default.Item, null, "Sub item 2"), _react.default.createElement(_index.default.Item, null, "Sub item 3 with sub", _react.default.createElement(_index.default.Submenu, null, _react.default.createElement(_index.default.Item, null, "Sub Sub 1"), _react.default.createElement(_index.default.Item, null, "Sub Sub 2"))))), _react.default.createElement(_index.default.Item, null, "Item 5"), _react.default.createElement(_index.default.Item, null, "Item 6")));
};

exports.OpenProgrammatically = OpenProgrammatically;

var CloseOnItemClick = function CloseOnItemClick() {
  var ref = (0, _react.useRef)(null);

  function onClick(e) {
    if (ref.current) {
      ref.current.toggle(e);
    }
  }

  return _react.default.createElement("div", {
    style: {
      marginLeft: 400
    }
  }, _react.default.createElement(_index.default, {
    title: "Dropdown example",
    ref: ref
  }, _react.default.createElement(_index.default.Item, {
    onClick: onClick
  }, "Close on Click"), _react.default.createElement(_index.default.Item, null, "Item 2"), _react.default.createElement(_index.default.Item, null, "Item 3"), _react.default.createElement(_index.default.Item, null, "Item 4 with sub", _react.default.createElement(_index.default.Submenu, null, _react.default.createElement(_index.default.Item, null, "Sub item 1 with sub", _react.default.createElement(_index.default.Submenu, null, _react.default.createElement(_index.default.Item, null, "Sub Sub 1"), _react.default.createElement(_index.default.Item, null, "Sub Sub 2"))), _react.default.createElement(_index.default.Item, null, "Sub item 2"), _react.default.createElement(_index.default.Item, null, "Sub item 3 with sub", _react.default.createElement(_index.default.Submenu, null, _react.default.createElement(_index.default.Item, null, "Sub Sub 1"), _react.default.createElement(_index.default.Item, null, "Sub Sub 2"))))), _react.default.createElement(_index.default.Item, null, "Item 5"), _react.default.createElement(_index.default.Item, null, "Item 6")));
};

exports.CloseOnItemClick = CloseOnItemClick;

var OpenOnHover = function OpenOnHover() {
  return _react.default.createElement("div", {
    style: {
      marginLeft: 400
    }
  }, _react.default.createElement(_index.default, {
    title: "Dropdown example",
    openOnHover: true
  }, _react.default.createElement(_index.default.Item, null, "Close on Click"), _react.default.createElement(_index.default.Item, null, "Item 2"), _react.default.createElement(_index.default.Item, null, "Item 3"), _react.default.createElement(_index.default.Item, null, "Item 4 with sub", _react.default.createElement(_index.default.Submenu, null, _react.default.createElement(_index.default.Item, null, "Sub item 1 with sub", _react.default.createElement(_index.default.Submenu, null, _react.default.createElement(_index.default.Item, null, "Sub Sub 1"), _react.default.createElement(_index.default.Item, null, "Sub Sub 2"))), _react.default.createElement(_index.default.Item, null, "Sub item 2"), _react.default.createElement(_index.default.Item, null, "Sub item 3 with sub", _react.default.createElement(_index.default.Submenu, null, _react.default.createElement(_index.default.Item, null, "Sub Sub 1"), _react.default.createElement(_index.default.Item, null, "Sub Sub 2"))))), _react.default.createElement(_index.default.Item, null, "Item 5"), _react.default.createElement(_index.default.Item, null, "Item 6")));
};

exports.OpenOnHover = OpenOnHover;
