// react-multilevel-dropdown/Dropdown.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Dashed = exports.SpecialSuccess = exports.Special = exports.Tertiary = exports.Secondary = exports.Primary = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _index = _interopRequireDefault(require("./index"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'Dropdown'
};
exports.default = _default;

var Items = function Items() {
  return _react.default.createElement(_react.default.Fragment, null, _react.default.createElement(_index.default.Item, null, "Item 1"), _react.default.createElement(_index.default.Item, null, "Item 2"), _react.default.createElement(_index.default.Item, null, "Item 3"), _react.default.createElement(_index.default.Item, null, "Item 4 with sub", _react.default.createElement(_index.default.Submenu, null, _react.default.createElement(_index.default.Item, null, "Sub item 1 with sub", _react.default.createElement(_index.default.Submenu, null, _react.default.createElement(_index.default.Item, null, "Sub Sub 1"), _react.default.createElement(_index.default.Item, null, "Sub Sub 2"))), _react.default.createElement(_index.default.Item, null, "Sub item 2"), _react.default.createElement(_index.default.Item, null, "Sub item 3 with sub", _react.default.createElement(_index.default.Submenu, null, _react.default.createElement(_index.default.Item, null, "Sub Sub 1"), _react.default.createElement(_index.default.Item, null, "Sub Sub 2"))))), _react.default.createElement(_index.default.Item, null, "Item 5"), _react.default.createElement(_index.default.Item, null, "Item 6"));
};

var Default = function Default() {
  return _react.default.createElement("div", {
    style: {
      marginLeft: 600
    }
  }, _react.default.createElement(_index.default, {
    title: "Dropdown example"
  }, _react.default.createElement(Items, null)));
};

exports.Default = Default;

var Primary = function Primary() {
  return _react.default.createElement("div", {
    style: {
      marginLeft: 600
    }
  }, _react.default.createElement(_index.default, {
    title: "Dropdown example",
    buttonVariant: "primary"
  }, _react.default.createElement(Items, null)));
};

exports.Primary = Primary;

var Secondary = function Secondary() {
  return _react.default.createElement("div", {
    style: {
      marginLeft: 600
    }
  }, _react.default.createElement(_index.default, {
    title: "Dropdown example",
    buttonVariant: "secondary"
  }, _react.default.createElement(Items, null)));
};

exports.Secondary = Secondary;

var Tertiary = function Tertiary() {
  return _react.default.createElement("div", {
    style: {
      marginLeft: 600
    }
  }, _react.default.createElement(_index.default, {
    title: "Dropdown example",
    buttonVariant: "tertiary"
  }, _react.default.createElement(Items, null)));
};

exports.Tertiary = Tertiary;

var Special = function Special() {
  return _react.default.createElement("div", {
    style: {
      marginLeft: 600
    }
  }, _react.default.createElement(_index.default, {
    title: "Dropdown example",
    buttonVariant: "special"
  }, _react.default.createElement(Items, null)));
};

exports.Special = Special;

var SpecialSuccess = function SpecialSuccess() {
  return _react.default.createElement("div", {
    style: {
      marginLeft: 600
    }
  }, _react.default.createElement(_index.default, {
    title: "Dropdown example",
    buttonVariant: "special-success"
  }, _react.default.createElement(Items, null)));
};

exports.SpecialSuccess = SpecialSuccess;

var Dashed = function Dashed() {
  return _react.default.createElement("div", {
    style: {
      marginLeft: 600
    }
  }, _react.default.createElement(_index.default, {
    title: "Dropdown example",
    buttonVariant: "dashed"
  }, _react.default.createElement(Items, null)));
};

exports.Dashed = Dashed;
